### ISSUE TITLE
no-redundant-vars to fix Local variable is redundant 

### ISSUE THREAD

**Issue by zhangKunUserGit** [eslint/eslint#16169]
### Rule details

Local variable is redundant

### Related ECMAScript feature

ES6

### What type of rule is this?

Warns about a potential problem

### Example code

```js
function getRawSvgElement(wrapper) {
  const iconHtml = wrapper.find('span').html();
  const svgElement = stringToDom(iconHtml).querySelector('svg');

  return svgElement;
}
```

### Why should this rule be in the core instead of a plugin?

<img width="691" alt="image" src="https://user-images.githubusercontent.com/13297317/181906425-6e492bef-6239-4715-badf-b98ec3dc6213.png">

Local variable 'svgElement' is redundant。Can eslint be added to this check。At present, idea did it。

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

**Comment by mdjermanovic** [comment]
Hi @zhangKunUserGit, thanks for the issue!

This is a stylistic rule that isn't related to new ECMAScript features that reached stage 4 in the preceding 12 months, so per our [rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies/) it doesn't qualify for inclusion in the core. You can always make a custom rule and publish a plugin.
